### ISSUE TITLE
URL Redirect does not work

### ISSUE THREAD

**Issue by ktomann** [mixmint/opnsense-captive-portal-template#19]
**Describe the bug**
URL redirect does not work after successful login

**To Reproduce**
1. Try to log in
2. it should redirect to https://guest.mydomain.tld:8000 (or another domain)

**Expected behavior**
Redirect after successful login

**Screenshots**

**Desktop (please complete the following information):**
- All browsers (Mobile and Desktop)

**Additional context**
Maybe I just can't find the right section where I can enter my URL.

**Comment by mixmint** [comment]
@ktomann I checked it myself, and everything works correctly - with the call via the http and https protocol. The difference in https is that it is necessary to have a correct certificate corresponding to the domain specified in the Captive Portal configuration.

Login also works correctly without redirection and with redirection triggered by CP user requests, the redirection usually looks like this:

`https://opnsense.localdomain:8000/index.html?redirurl=https://google.com`

or more simply

`https://opnsense.localdomain:8000/?redirurl=https://google.com`

I think you have a **problem with the SSL certificate** or you incorrectly entered the **required domain in CP**, which does not correspond to the one in the certificate. In such a case, you usually get a redirection loop until it stops and then you can assume that CP did not redirect anywhere.

BTW - the exception described above has nothing to do with the correct operation of the CP template

**Comment by ktomann** [comment]
Hi @mixmint 

Thanks for your reply.
I think I described it wrong.

When I connect to the guest network, I am immediately redirected to the CP. The certificate is valid and everything is displayed correctly.

However, after a successful login in the CP I would like to be redirected to any page, or to the page of the CP, so that the guest could log out again or see his remaining time.

At the moment, the pop-up window from the CP closes immediately and I am online.

Do I have to activate the redirection somewhere?

Many thanks for your help

Best, KrishnaT

**Comment by mixmint** [comment]
I don't think so, CP in OPNsense performs redirects to "itself" by default and the only determinant of where it should redirect outside of itself is the GET parameter "redirurl=" according to the following condition:

```
if ($.getUrlparams()['redirurl'] != undefined) {
	window.location = $.getUrlparams()['redirurl'] + '?refresh';
} else {
	window.location.reload();
}
```

However, what you are writing about looks like the browser is responsible for such window behavior. Please specify in more detail how the browser opens the login portal, is it a pop-up window without a navigation bar? Can you refresh it with Ctrl+F5 before logging in?
